tests/dialects/test_x86.py

```diff
@@ -27,3 +28,11 @@
 def test_register(register: x86.register.GeneralRegisterType, name: str):
     assert register.is_allocated
     assert register.register_name == name
+
+
+def test_add_op():
+    r1 = TestSSAValue(x86.register.RAX)
+    r2 = TestSSAValue(x86.register.RDX)
+    add_op = x86.AddOp(r1, r2, result=x86.register.RAX)
+
+    assert add_op.assembly_line() == "    add rax, rdx"
```

Comment:
This is already tested in the filecheck, we don't need an extra pytest here

Reply:
I don't think filecheck tests the assembly printing?

Reply:
ah yes then let's add a filecheck for that, like for riscv :)